web-ui/src/pages/KudosPage.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { styled } from "@mui/material/styles";
+import { Button, Tab, Typography } from "@mui/material";
+import { TabContext, TabList, TabPanel } from "@mui/lab";
+import { AppContext } from "../context/AppContext";
+import { selectCsrfToken, selectCurrentUser } from "../context/selectors";
+import { getReceivedKudos, getSentKudos } from "../api/kudos";
+import { UPDATE_TOAST } from "../context/actions";
+import KudosCard from "../components/kudos_card/KudosCard";
+
+import "./KudosPage.css";
+import KudosDialog from "../components/kudos_dialog/KudosDialog";
+import StarIcon from "@mui/icons-material/Star";
+
+import ArchiveIcon from "@mui/icons-material/Archive";
+import UnarchiveIcon from "@mui/icons-material/Unarchive";
+
+import SkeletonLoader from "../components/skeleton_loader/SkeletonLoader";
+
+const PREFIX = "KudosPage";
+const classes = {
+  expandOpen: `${PREFIX}-expandOpen`,
+  expandClose: `${PREFIX}-expandClose`,
+};
+
+const Root = styled("div")({
+  [`& .${classes.expandOpen}`]: {
+    transform: "rotate(180deg)",
+    transition: "transform 0.1s linear",
+    marginLeft: "auto",
+  },
+  [`& .${classes.expandClose}`]: {
+    transform: "rotate(0deg)",
+    transition: "transform 0.1s linear",
+    marginLeft: "auto",
+  },
+});
+
+const KudosPage = () => {
+  const { state, dispatch } = useContext(AppContext);
+  const csrf = selectCsrfToken(state);
+
+  const currentUser = selectCurrentUser(state);
+
+  const [kudosDialogOpen, setKudosDialogOpen] = useState(false);
+  const [kudosTab, setKudosTab] = useState("RECEIVED");
+  const [receivedKudos, setReceivedKudos] = useState([]);
+  const [receivedKudosLoading, setReceivedKudosLoading] = useState(true);
+  const [sentKudos, setSentKudos] = useState([]);
+  const [sentKudosLoading, setSentKudosLoading] = useState(true);
+
+  const loadReceivedKudos = useCallback(async () => {
+    setReceivedKudosLoading(true);
+    const res = await getReceivedKudos(currentUser.id, csrf);
+    if (res?.payload?.data && !res.error) {
+      setReceivedKudosLoading(false);
+      return res.payload.data;
+    } else {
```

Comment:
Lines 58-65 can be deleted because `getReceivedKudos` calls `resolve` which provides all the error handling needed.